### PR TITLE
The infinite-capacity power cell is now closer to infinity

### PR DESCRIPTION
**Pull request by yackemflam**
:cl: 
* rscadd: INFINITE POWER

### CODE REVIEW

**Comment by Exxion** [comment]
But it is legitimately infinite already because of its overridden `use()`

**Comment by yackemflam** [comment]
It's for functions that are based off of a powercells charge.

File: code/game/objects/items/weapons/power_cells.dm [hunk 1]
```diff
@@ -95,7 +95,7 @@
 	name = "infinite-capacity power cell!"
 	icon_state = "icell"
 	origin_tech =  null
-	maxcharge = 30000
+	maxcharge = 1000000000 //Good luck charging this, buff it if it gets fully charged
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/power_cells.dm:98]: You could use defines for this instead, such as

`maxcharge = ARBITRARILY_LARGE_NUMBER * 100000`

Comment (yackemflam) [reply code/game/objects/items/weapons/power_cells.dm:98]: Makes no real difference

Comment (PJB3005) [reply code/game/objects/items/weapons/power_cells.dm:98]: I think he was joking.

REAL TALK: `maxcharge = 1.#INF`

**Comment by Exxion** [comment]
Ah
Yeah make it actually infinite
There's an `INFINITY` define or something

File: code/game/objects/items/weapons/power_cells.dm (hunk 1 above)
Comment (troots) [reply code/game/objects/items/weapons/power_cells.dm:98]: `maxcharge = ∞`
this powercell's charging ports are infinity symbol shaped

**Comment by PJB3005** [comment]
@Exxion `INFINITY` is just a large number.

**Comment by SonixApache** [comment]
but why

File: code/game/objects/items/weapons/power_cells.dm (hunk 1 above)
Comment (ComicIronic) [reply code/game/objects/items/weapons/power_cells.dm:98]: Can't you just use the INFINITY define?

**Comment by PJB3005** [comment]
Actually, on bay it's `#define INFINITY 1.#INF`, and I accidentally ported that in the XGM branch: https://github.com/vgstation-coders/vgstation13/blob/8266b35e24440a9ded49528c35590baa85bb0202/__DEFINES/math_physics.dm#L33

File: code/game/objects/items/weapons/power_cells.dm (hunk 1 above)
Comment (RemieRichards) [reply code/game/objects/items/weapons/power_cells.dm:98]: @ComicIronic that's a custom define, it's just a big number, smaller than PJB's ```1.#INF```

really though the ```use()``` or ```use_power()``` or however functions just returning ```TRUE``` is proper "infinity" power.

Comment (ComicIronic) [reply code/game/objects/items/weapons/power_cells.dm:98]: Wait, infinity isn't built into the language?

Comment (RemieRichards) [reply code/game/objects/items/weapons/power_cells.dm:98]: It is, as ```1.#INF```, but the ```INFINITY``` value is just a ```#define``` that was made ages ago, it's not a built-in.

Comment (PJB3005) [reply code/game/objects/items/weapons/power_cells.dm:98]: @RemieRichards pretty sure having a `use()` that's overridden is already how the infinite power cell works.

Comment (RemieRichards) [reply code/game/objects/items/weapons/power_cells.dm:98]: ya, the PR is just for memes, that's my point.

Comment (Kurfursten) [reply code/game/objects/items/weapons/power_cells.dm:98]: Yeah, it was a joke about how arbitrarily large number is much smaller than the number you want

I am amused about the defines discussion though

**Comment by Yelier** [comment]
Will spacepods with it finally be able to go forever?

**Comment by PJB3005** [comment]
Sounds like space pod code needs to learn what a private variable is.

**Comment by Exxion** [comment]
Sounds like BYOND needs to learn what a private variable is

**Comment by clusterfack** [comment]
Cant believe you guys let this completely pointless shit go on for this long

**Pull request closed without merging**